File: src/main/java/com/pascalnb/dbwrapper/Query.java
package com.pascalnb.dbwrapper;

import org.jetbrains.annotations.Contract;

/**
 * Class that represents a database query with options to insert values.
 */
@SuppressWarnings("unused")
public class Query {

    private final String query;
    private final Object[] args;

    public Query(String query, Object... args) {
        this.query = query;
        this.args = args;
    }

    @Contract("_ -> new")
    public Query withArgs(Object... args) {
        return new Query(query, args);
    }

    @Override
    public String toString() {
        return query;
    }

    public Object[] getArgs() {
        return args;
    }

}

File: src/main/kotlin/com/pascalnb/dbwrapper/Query.kt
package com.pascalnb.dbwrapper

import org.jetbrains.annotations.Contract

/**
 * Class that represents a database query with options to insert values.
 *
 * @param query the query
 */
class Query(private val query: String, vararg args: Any?) {
    val args: Array<Any?>

    init {
        this.args = arrayOf(*args)
    }

    @Contract("_ -> new")
    fun withArgs(vararg args: Any?): Query {
        return Query(query, *args)
    }

    override fun toString(): String {
        return query
    }
}